File: 프로그래머스/lv0/120824. 짝수 홀수 개수/짝수 홀수 개수.py
def solution(num_list):
    # javascript reduce같은 문법! 
    evenCnt = 0 # 짝수 개수
    oddCnt = 0
    for i in num_list:
        if i % 2 == 0:
            evenCnt += 1
        else:
            oddCnt += 1
    
    return [evenCnt, oddCnt]

File: 프로그래머스/lv0/120824. 짝수 홀수 개수/짝수 홀수 개수.js
const isWholeNumbers = (arr) => arr.some(e => typeof e !== 'number')

const getCountOfEvenNumber = (arr) => {
    if(isWholeNumbers(arr)) throw new Error("숫자가 아닌 값이 있습니다!")
    return arr.filter(e => e % 2 === 0).length;
}
function solution(num_list) {
    return [getCountOfEvenNumber(num_list), num_list.length - getCountOfEvenNumber(num_list)];
}